### PR TITLE
Make netcat-openbsd the default netcat

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

See Issue #19411. 

TL;DR: netcat-gnu does neither support IPv6 nor unix sockets and is therefore not a good idea for a default netcat.
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Fixed and tested compilation of all pkgs that depend on this change
  - [x] NixOS tests
    - [x] `nixos/tests/hibernate.nix`
    - [x] `nixos/tests/virtualbox.nix`
  - [x] playonlinux
  - [x] xfce4-sensors-plugin
  - [ ] `build-support/vm/windows`
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I am open for suggestions on, how to test the windows vm build-support with GNU netcat.

---

A problem I see here is, that there is no standard deprecation process in nixpkgs (#15357). This change would certainly benefit from a `warnSemanticChange` which I proposed in this PR: #19315.

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -2795,7 +2795,9 @@ in
 
   netboot = callPackage ../tools/networking/netboot {};
 
-  netcat = callPackage ../tools/networking/netcat { };
+  netcat = netcat-openbsd;
+
+  netcat-gnu = callPackage ../tools/networking/gnu-netcat { };
```
Comment (vcunat) [inline pkgs/top-level/all-packages.nix]: Nitpick: the naming is rather inconsistent. I think it would be better to use `netcat-gnu` even for the directory name.

Comment (sternenseemann) [inline pkgs/top-level/all-packages.nix]: Yeah, I had named it differently before, good point

Comment (sternenseemann) [inline pkgs/top-level/all-packages.nix]: Fixed

File: nixos/tests/hibernate.nix [hunk 2]
```diff
@@ -36,7 +36,7 @@ import ./make-test.nix (pkgs: {
       $machine->waitForShutdown;
       $machine->start;
       $probe->waitForUnit("network.target");
-      $probe->waitUntilSucceeds("echo test | nc -c machine 4444");
+      $probe->waitUntilSucceeds("echo test | nc -q 0 machine 4444");
```
Comment (Mic92) [inline nixos/tests/hibernate.nix]: Why is the timeout here needed?

Comment (sternenseemann) [inline nixos/tests/hibernate.nix]: This is the equivalent to `-c` in netcat-openbsd. I couldn't find another way to accomplish closing at `EOF`. I also could not override the `systemPackages` to include netcat-gnu instead of netcat-openbsd.

Comment (sternenseemann) [inline nixos/tests/hibernate.nix]: Oh, it is not needed indeed, let me fix that.

Comment (sternenseemann) [inline nixos/tests/hibernate.nix]: Done

**Comment by sternenseemann** [comment]
\o/
